library/std/src/os/solid/io.rs

```diff
@@ -27,10 +27,12 @@ pub trait FromRawFd {
     /// Constructs a new instance of `Self` from the given raw file
     /// descriptor.
     ///
-    /// This function **consumes ownership** of the specified file
+    /// This function **consumes [ownership][io-safety]** of the specified file
```

Comment:
issue: this isn't quite correct, it *may* consume ownership. It depends on the type being used.

I think the safe thing to say here is that by default it consumes ownership, but in some cases it may not if explicitly stated (mention the Stdio types), and in the FromRawFd docs of the Stdio types, add a note saying that it does not consume ownership

Reply:
The stdio types don't implement implement `FromRawFd`. There is one type in std which does and doesn't consume ownership, and it's `RawFd`.

I was unaware that SOLID had its own copy of this text; the [main `from_raw_fd`](https://doc.rust-lang.org/stable/std/os/fd/trait.FromRawFd.html#tymethod.from_raw_fd) says "typically consumes ownership", so it looks like we need to sync up these two copies.

Reply:
I think I can also just remove the copy in SOLID and Fortanix, and then the default trait text will be used by rustdoc.

Reply:
Ah no wait they have their own copy of the trait. I'll just copy the text from the standard trait.